### ISSUE TITLE
Create a @RecommendedColumn annotation

### ISSUE THREAD

**Issue by isabelle-dr** [MobilityData/gtfs-validator#1350]
## Context
The spec requires certain columns, but having an empty value in them is valid (example [transfers.transfer_type](https://gtfs.org/schedule/reference/#transferstxt)).
The fact that we only had the `@Required` annotation, used in the schema files created [false positives](https://github.com/MobilityData/gtfs-validator/issues/1198).
```
  @Required
  GtfsTransferType transferType();
```
This is why the awesome @briandonahue created a new @RequiredColumn annotation in #1344, designed especially for the case where the column itself is required, but some of the fields can be empty.
```
  @RequiredColumn
  GtfsTransferType transferType();
```

## Proposition
We have a similar situation with `stop_times.timepoint`, except it's not a requirement but rather a recommendation. The GTFS Best Practice mentions (EDIT: they mentioned, it looks like we had another accidental removal in https://github.com/MobilityData/GTFS_Schedule_Best-Practices/pull/44, I am working on it 😅) that the timepoint should be provided, although the values can be empty under certain conditions.

In the validator, we have a [missing_timepoint_column](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#missing_timepoint_column) notice that looks for the presence of the column
For consistency with the work done in #1344, we could replace this notice with a `missing_recommended_column` using a `@RecommendedColumn`. 

## Alternatives considered
Keep the validator as-is. I acknowledge this item doesn't add much to the user experience but figured this validator might be easier to maintain if we made this consistent (developers, I would love your POV). 
Additionally, we might want to use the `@RecommendedColumn` for other fields than `stop_times.timepoint` in the future.

**Comment by briandonahue** [comment]
@isabelle-dr This seems relevant to an issue I created, #1343.  I will respond there and discuss a few more scenarios.

**Comment by davidgamez** [comment]
@isabelle-dr, let's review the list of scenarios will be nice to migrate from `@Required` to `@RecommendedColumn`, including the proposals from @briandonahue https://github.com/MobilityData/gtfs-validator/issues/1343#issuecomment-1466302162

**Comment by isabelle-dr** [comment]
@davidgamez here is the notice change we can do with the `@RecommendedColumn` annotation.
This goes along the lines of the [`missing_required_column`](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#missing_required_column) notice triggered by the @RequiredColumn annotation.
- Remove [`missing_timepoint_column`](https://github.com/MobilityData/gtfs-validator/blob/master/RULES.md#missing_timepoint_column) from the validator and replace by a `missing_recommended_column` notice using the `@RecommendedColumn` in `stop_times.timepoint`.
